src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist

```diff
@@ -137,6 +137,8 @@ staticMethod java.lang.String valueOf java.lang.Object
 staticMethod java.lang.System currentTimeMillis
 method java.lang.Throwable getCause
 method java.lang.Throwable getMessage
+method java.lang.Throwable printStackTrace java.io.PrintStream
+method java.lang.Throwable printStackTrace java.io.PrintWriter
```

Comment:
I didn't know about `catchError` (thanks for pointing it out), but I just tested it and I can see that it does log the exception stacktrace so would have been acceptable in most cases except that it doesn't work correctly. E.g., I need to set the result to `UNSTABLE` instead of `FAILED` and also need to log a troubleshooting message, but for some reason, none of these parameters are getting recognized. E.g., I tried the below:

```
def F() {
    catchError(buildResult: 'UNSTABLE', stageResult: 'UNSTABLE', message: 'Got unexpected exception') {
        throw new Exception("Testing catchError")
    }
}

stage('Test') {
    F()
}
```

Both the stage and build result are still set to `FAILED` and the `message` is nowhere to be found. Could this be a bug?

However, it is still very useful to have these signatures whitelisted because, using using `try/catch/finally` could make the code look more natural in a scripted approach instead of using `catchError` and in fact is already pointed out in the documentation here: https://jenkins.io/doc/pipeline/steps/workflow-basic-steps/#catcherror-catch-error-and-set-build-result-to-failure.

Reply:
@dwnusbaum I am just wondering if you are waiting for any more inputs from me on this review?

Reply:
> Both the stage and build result are still set to FAILED and the message is nowhere to be found.

Weird, what version of Pipeline: Basic Steps Plugin and Pipeline: Groovy do you have installed?

Reply:
But yeah, I see how these signatures could be useful in scripted, so I'm not totally opposed to it, I just wanted to check if `catchError` handles your use case.

Reply:
Thanks for approving! Here is the version information you asked:

Pipeline: Basic Steps: 2.15 (latest is 2.18)
Pipeline: Groovy: 2.70 (latest is 2.78)

Reply:
Ok, thanks for the update, you need [Pipeline: Basic Steps 2.16+](https://wiki.jenkins.io/display/JENKINS/Pipeline+Basic+Steps+Plugin) for the new options to `catchError` and [Pipeline: Groovy 2.73+](https://github.com/jenkinsci/workflow-cps-plugin/blob/master/CHANGELOG.md#273) for the fix that logs a warning when you pass unsupported options to a step.